unified-runtime/source/adapters/hip/program.cpp

```diff
@@ -546,16 +538,15 @@ UR_APIEXPORT ur_result_t UR_APICALL urProgramGetFunctionPointer(
   hipFunction_t Func;
   hipError_t Ret = hipModuleGetFunction(&Func, hProgram->get(), pFunctionName);
   *ppFunctionPointer = Func;
-  ur_result_t Result = UR_RESULT_SUCCESS;
-
   if (Ret != hipSuccess && Ret != hipErrorNotFound)
-    UR_CHECK_ERROR(Ret);
+    return mapErrorUR(Ret);
+
   if (Ret == hipErrorNotFound) {
```

Comment:
This is a nit, but if we check for `hipErrorNotFound` first, we wouldn't need to re-check for it in the second `if`. Maybe tying them into `if`/`else if` would be nicer?